api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AssetGroupServiceImpl.java

```diff
@@ -857,4 +836,45 @@ private String getPluginAliasQuery(String datasource) {
 		}
 		return "";
 	}
+
+	@Override
+	public boolean deleteAssetGroupByGroupName(String groupName) {
+		try {
+			AssetGroupDetails assetGroupInDB = assetGroupRepository.findByGroupName(groupName);
+			if (assetGroupInDB == null) {
+				log.error("Asset group not found with group name as {}", groupName);
+				return false;
+			}
+			assetGroupRepository.delete(assetGroupInDB);
+			log.info("asset group with group name {} deleted successfully", groupName);
+			return true;
+		} catch (Exception e) {
+			log.error("Unable to delete asset group with group name as {}", groupName, e);
+			return false;
+		}
+	}
+
+	@Override
+	public void updatePluginDataInAssetGroup(String pluginName, boolean isRemoved) {
+		try {
+			StringBuilder requestBody = new StringBuilder("{\"actions\":[");
+			if (isRemoved) {
+				requestBody.append("{\"remove\":{\"index\":\"").append(pluginName)
+						.append("_*\",\"alias\":\"").append(ALL_SOURCES_ASSET_GROUP).append("\"}},");
+			} else {
+				requestBody.append("{\"add\":{\"index\":\"").append(pluginName)
+						.append("_*\",\"alias\":\"").append(ALL_SOURCES_ASSET_GROUP).append("\"}},");
+			}
+			requestBody.deleteCharAt(requestBody.length() - 1);
+			requestBody.append("]}");
+			Response response = commonService.invokeAPI("POST", ES_ALIASES_PATH, requestBody.toString());
+			if (response == null || response.getStatusLine().getStatusCode() != 200) {
+				log.error("Unable to remove or update indices for asset group {} with response {}",
+						ALL_SOURCES_ASSET_GROUP, response);
+			}
+			log.info("Indices updated for plugin successfully.");
```

Comment:
For which plugin?
"Indices for {} updated.

Reply:
updated